modules/storage-aws-s3/s3/s3.go

```diff
@@ -72,19 +72,11 @@ func makeFilePath(parts ...string) string {
 }
 
 func (s *s3) StoreSnapshot(ctx context.Context, snapshot *snapshots.Snapshot) error {
-	// create bucket
-	bucketName := s.config.BucketName()
-	bucketExists, err := s.client.BucketExists(ctx, bucketName)
+	bucketName, err := s.findBucket(ctx)
 	if err != nil {
-		return snapshots.NewErrInternal(errors.Wrap(err, "check bucket exists"))
-	}
-	if !bucketExists {
-		err = s.client.MakeBucket(ctx, bucketName, minio.MakeBucketOptions{})
-		if err != nil {
-			return snapshots.NewErrInternal(
-				errors.Wrapf(err, "create bucket '%s'", bucketName))
-		}
+		return snapshots.NewErrInternal(errors.Wrap(err, "store snapshot"))
```

Comment:
I would change the error to be more precise. I would change the message to "find bucket"